### PR TITLE
CMR-10212: SNS/SQS testing infrastructure

### PR DESCRIPTION
**Pull request by eereiter**
# Overview

### What is the feature/fix?

Adding SNS/SQS testing infrastructure

### What is the Solution?

Adding SNS/SQS testing infrastructure - setting up elasticMQ

### What areas of the application does this impact?

SNS/SQS - new feature.

# Checklist

- [X] I have updated/added unit and int tests that prove my fix is effective or that my feature works
- [X] New and existing unit and int tests pass locally and remotely
- [X] clj-kondo has been run locally and all errors corrected
- [X] I have removed unnecessary/dead code and imports in files I have changed
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have cleaned up integration tests by doing one or more of the following:
  - migrated any are2 tests to are3 in files I have changed
  - de-duped, consolidated, removed dead int tests
  - transformed applicable int tests into unit tests
  - refactored to reduce number of system state resets by updating fixtures (use-fixtures :each (ingest/reset-fixture {})) to be :once instead of :each

### CODE REVIEW

File: message-queue-lib/src/cmr/message_queue/test/test_util.clj [hunk 1]
```diff
@@ -0,0 +1,28 @@
+(ns cmr.message-queue.test.test-util
+  "Namespace to test local sqs server"
+  (:require
+   ;[clj-http.lite.client :as client]
```
Comment (TylerHeald1) [inline message-queue-lib/src/cmr/message_queue/test/test_util.clj:4]: can be removed?

**Review by TylerHeald1 — APPROVED** [review]

File: message-queue-lib/src/cmr/message_queue/test/test_util.clj (hunk 1 above)
Comment (jmaeng72) [reply message-queue-lib/src/cmr/message_queue/test/test_util.clj:4]: +1

**Review by jmaeng72 — APPROVED** [review]

File: message-queue-lib/src/cmr/message_queue/test/test_util.clj (hunk 1 above)
Comment (eereiter) [reply message-queue-lib/src/cmr/message_queue/test/test_util.clj:4]: Fixed.

File: message-queue-lib/src/cmr/message_queue/topic/local_topic.clj [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+(defn publish-message
+  "Go through the subscriptions and publish the message to each queue that
+  has a filter that matches the message-attributes. If the queue doesn't have
+  a filter, then pass the message to them. Returns a list of AWS publish responses."
+  [subscription message message-attributes]
+  (let [{:keys [sqs-client filter queue-url dead-letter-queue-url]} subscription
+        msg-atts (queue/attributes-builder message-attributes)]
+    (try
+      (if filter
+        (when (or (and (= (:collection-concept-id message-attributes)
```
[the remark this answers is not in the document]
Comment (eereiter) [reply message-queue-lib/src/cmr/message_queue/topic/local_topic.clj:22]: Yes, I have updated the code.

**Pull request merged**